faiss/impl/RaBitQuantizer.cpp

```diff
@@ -0,0 +1,487 @@
+#include <faiss/impl/RaBitQuantizer.h>
+
+#include <algorithm>
+#include <cmath>
+#include <cstring>
+#include <limits>
+#include <memory>
+#include <vector>
+
+#include <faiss/impl/FaissAssert.h>
+#include <faiss/impl/platform_macros.h>
+#include <faiss/utils/distances.h>
+
+namespace faiss {
+
+struct FactorsData {
+    // ||or - c||
+    float factor_0 = 0;
```

Comment:
why not give the fields proper names? Otherwise you might as well use `float factors[4]`